pyiron_atomistics/atomistics/structure/strain.py

```diff
@@ -0,0 +1,169 @@
+        sign = np.sign(np.sum(np.cross(v, w)*vec_axis, axis=-1))
+        vec_axis *= (sign*self._get_angle(v, w)/4)[:,None]
+        return Rotation.from_mrp(vec_axis).as_matrix()
+
+    @property
+    def rotations(self):
+        """Rotation for each atom to find the correct pairs of coordinates."""
+        if self._rotations is None:
+            v = self.coords.copy()[:,0,:]
+            w_first = self.ref_coord[
+                np.linalg.norm(self.ref_coord[None,:,:]-v[:,None,:], axis=-1).argmin(axis=1)
+            ].copy()
+            first_rot = self._get_rotation_from_vectors(v, w_first)
+            all_vecs = np.einsum('nij,nkj->nki', first_rot, self.coords)
+            highest_angle_indices = np.absolute(
+                np.sum(all_vecs*all_vecs[:,:1], axis=-1)
+            ).argmin(axis=-1)
+            v = all_vecs[np.arange(len(self.coords)),highest_angle_indices,:]
+            dv = self.ref_coord[None,:,:]-v[:,None,:]
+            dist = np.linalg.norm(dv, axis=-1)
+            dist += np.absolute(np.sum(dv*all_vecs[:,:1], axis=-1))
+            w_second = self.ref_coord[dist.argmin(axis=1)].copy()
+            second_rot = self._get_rotation_from_vectors(v, w_second, all_vecs[:,0])
+            self._rotations = np.einsum('nij,njk->nik', second_rot, first_rot)
+        return self._rotations
+
+    @staticmethod
+    def _get_best_match_indices(coords, ref_coord):
+        distances = np.linalg.norm(coords[:,:,None,:]-ref_coord[None,None,:,:], axis=-1)
+        return np.argmin(distances, axis=-1)
+
+    @staticmethod
+    def _get_majority_phase(structure):
+        cna = structure.analyse.pyscal_cna_adaptive()
+        return np.asarray([k for k in cna.keys()])[np.argmax([v for v in cna.values()])]
+
+    @staticmethod
+    def _get_number_of_neighbors(crystal_phase):
+        if crystal_phase=='bcc':
+            return 8
+        elif crystal_phase=='fcc' or crystal_phase=='hcp':
+            return 12
+        else:
+            raise ValueError('Crystal structure not recognized')
+
+    @property
+    def ref_coord(self):
+        """Reference local coordinates."""
+        if self._ref_coord is None:
+            self._ref_coord = self.ref_structure.get_neighbors(
+                num_neighbors=self.num_neighbors
+            ).vecs[0]
+        return self._ref_coord
+
+    @property
+    def coords(self):
+        """Local coordinates of each atom."""
+        if self._coords is None:
+            self._coords = self.structure.get_neighbors(num_neighbors=self.num_neighbors).vecs
+        return self._coords
+
+    @property
+    def _indices(self):
+        all_vecs = np.einsum('nij,nkj->nki', self.rotations, self.coords)
+        return self._get_best_match_indices(all_vecs, self.ref_coord)
+
+    @property
+    def strain(self):
+        """Strain value of each atom"""
+        D = np.einsum('ij,ik->jk', self.ref_coord, self.ref_coord)
+        D = np.linalg.inv(D)
```

Comment:
```suggestion
        Dinverse = np.einsum('ij,ik->jk', self.ref_coord, self.ref_coord)
        D = np.linalg.inv(Dinverse)
```

The item in the first definition is the inverse of D, so let's just name it that. I guess this is slightly worse for performance since there's an extra variable kicking around, but its truthier to read and I trust the garbage collection to have our backs.